### ISSUE TITLE
Is this project still alive?

### ISSUE THREAD

**Issue by skyline75489** [MiniKeePass/MiniKeePass#486]
It's been quite long time since this project received last update.

**Comment by skyline75489** [comment]
Any way I can contact the author? Anyone?...

**Comment by benjibobs** [comment]
This is open source under GNU 3 and as such if you wanted to fork it (to revive it) you're legally allowed to do so. 

The app works fine as is, so perhaps the author just feels no need to push any updates.

**Comment by timbru31** [comment]
@benjibobs 
> The app works fine as is, so perhaps the author just feels no need to push any updates.
It works, yes, but there is still no fix for the iPad layout bug, missing document picker support, no KDBX4 support, the cursor bug is still open. Just to name a few

**Comment by skyline75489** [comment]
@benjibobs Exactly. I'm trying to work on those issues but find that this project is no longer actively maintained.

**Comment by timbru31** [comment]
I'm currently examining of https://github.com/keeweb/keeweb works offline mode on iOS, too.
One could use https://app.keeweb.info, but I'd prefer to use a self hosted and offline capable variant.

Dropbox built in web browser supports JS, Nextcloud e.g. not (https://github.com/nextcloud/ios/issues/190), you might want to take a look at keeweb.

**Comment by benjibobs** [comment]
There are some pull requests addressing the issues you guys are discussing, I might fork this and merge those PRs to see if we could get an updated build distributed

**Comment by skyline75489** [comment]
I thought about forking or even creating a new project based on MiniKeePass. I hesitate to do so because one united community is always better than two (rivalry) ones. See what happens between NeoVim & Vim. Even though KeePass isn't widely used as Vim, I'd really like to see all of us focusing on one project.

**Comment by wooster11** [comment]
Your best bet would then be to try and contact the owners and see if you
can take over the active maintenance. It's likely a time and effort issue
which has led to the lack of progress.

**Comment by adrianmoisey** [comment]
> I thought about forking or even creating a new project based on MiniKeePass. I hesitate to do so because one united community is always better than two (rivalry) ones.

That's a valid point I agre with. However, sometimes a fork ends up being stronger than the original. See https://github.com/keepassxreboot/keepassxc for example.

**Comment by tssmith7** [comment]
I've been working on this code for a short bit, mostly as a learning experience.  I've managed to include KDBX4 support, TwoFish support for V1.x databases, and integrated dropbox file support.  

But, after a while (not knowing how to navigate Github), I noticed that the owners have most recently (Nov 2016) been working on converting the UI to Storyboards using swift and auto layout. It seems that maybe that is the direction they want to go to solve the device layout issues. 

I've put the Dropbox integration code in the auto-layout branch as it seems it would be most likely to be included in the code base. 

I would be interested in getting this project going again as well.

**Comment by timbru31** [comment]
@tssmith7 Nice! But even November is like half a year in the past where the last commit was made.

**Comment by benjibobs** [comment]
> even November is like half a year in the past 

Damn... time flies!

@tssmith7 I saw your fork... it's very much advanced compared to the others I've seen in terms of how much new stuff has been implemented - I'm going to be keeping an eye on it, nice work!

**Comment by 4brunu** [comment]
I share the same idea of @skyline75489.
It would be great if the owners would add someone that would manage the project.

**Comment by tssmith7** [comment]
I have gotten an initial working copy of my updates to MiniKeePass.  For anyone who wants to compile the sources themselves to run on your own phone you can now do the following: 
NOTE: You do not need to pay the $99 developer fee to run an app on your own phone.

Update to the latest MacOS 10.12.5
Update to Xcode 8.3.2
Make sure that `git` is installed.

From the Terminal Window run the following commands to create a folder called MiniKeePass that will contain all the sources needed.

prompt$    **git clone `https://github.com/tssmith7/MiniKeePass`** 
`Cloning into 'MiniKeePass'...
remote: Counting objects: 8889, done.
remote: Compressing objects: 100% (186/186), done.
remote: Total 8889 (delta 157), reused 168 (delta 86), pack-reused 8617
Receiving objects: 100% (8889/8889), 16.94 MiB | 9.22 MiB/s, done.
Resolving deltas: 100% (6102/6102), done.
`
prompt$ **cd MiniKeePass**
prompt$ **git submodule init**
`Submodule 'MBProgressHUD' (https://github.com/jdg/MBProgressHUD) registered for path 'MBProgressHUD'
Submodule 'dropbox-sdk-obj-c' (https://github.com/dropbox/dropbox-sdk-obj-c) registered for path 'dropbox-sdk-obj-c'
Submodule 'phc-winner-argon2' (https://github.com/P-H-C/phc-winner-argon2) registered for path 'phc-winner-argon2'
`
prompt$ **git submodule update**
`Cloning into '/Users/tssmith/Documents/Xcode/MiniKeePass/MBProgressHUD'...
Cloning into '/Users/tssmith/Documents/Xcode/MiniKeePass/dropbox-sdk-obj-c'...
Cloning into '/Users/tssmith/Documents/Xcode/MiniKeePass/phc-winner-argon2'...
Submodule path 'MBProgressHUD': checked out 'c843711aa8a6741ceb755a641b2b5c98297cb295'
Submodule path 'dropbox-sdk-obj-c': checked out 'ccdb30f1d33ab7cff9107fc139da4df711dfea88'
Submodule path 'phc-winner-argon2': checked out '9e3ae0f5e91eced6de3c28a9b61f2d91200119e3'
`

Now open MiniKeePass.xcodeproj in Xcode.  Connect your phone and select it as the device target for the MiniKeePass scheme and compile/run.

Dropbox will not work unless you setup your own API-KEY or send me your email address to add you to my API-KEY.

Let me know if anyone can get this to work....

**Comment by tallerthenyou** [comment]
Sorry for the lack of attention.

As you have all noticed from the state of the repository we haven't been giving MiniKeePass much attention for a while. @jarush and I have been idly working on a Swift/Autolayout conversion as @tssmith7 noticed, but even that has been lacking. I'm interested in trying to get a new version together based on the autolayout branch, but I'm hesitant to make such a big change for basically no user visible benefit. On the flip side I also don't want to add a bunch of other changes in addition to totally overhauling most of the code.

The changes to add KDB4 support could be a good addition that users will appreciate, but shouldn't disrupt any existing database support.

**Comment by 4brunu** [comment]
Maybe @tssmith7 can try to create a clean short and compact Pull Request with only the KDB4 support...
@tallerthenyou Would you be willing to merge some small Pull Request to address some issues with the app? Maybe have a list of tasks, someone could pick them up, and create small Pull Request, what do you think?
Thanks for your afford on keep making this project.

**Comment by timbru31** [comment]
Thanks for the response @tallerthenyou.  
>but I'm hesitant to make such a big change for basically no user visible benefit.

TestFlight (Apple's beta program) might help with this (to make sure the release is bug free)?

Since it seems that there are some more interested people than just you and @jarush, have you thought about adding more contributors to the project?

**Comment by skyline75489** [comment]
+1 on more contributors from community.

**Comment by bmhatfield** [comment]
@tallerthenyou Just wanted to share that I think the benefit to Autolayout may be larger - it may help with layout issues on various devices ([iPad landscape, for example](https://github.com/MiniKeePass/MiniKeePass/issues/433)). Plus, if it unblocks you from further development or adding folks like @tssmith7, it seems worth it :-)

**Comment by dermaaarkus** [comment]
Sounds good, would also like to contribute some stuff if we get the project alive again :)

**Comment by timbru31** [comment]
Thanks a lot @tallerthenyou for the update! Can this issue be closed now? Project seems more active again.
